tests/unit/lms/services/grouping_test.py

```diff
@@ -13,8 +14,8 @@ class TestGroupingService:
     CONTEXT_ID = "context_id"
     TOOL_CONSUMER_INSTANCE_GUID = "t_c_i_guid"
 
-    def test_upsert_with_parents_inserts(self, svc, db_session):
-        course = factories.Course()
+    def test_upsert_with_parents_inserts(self, svc, application_instance, db_session):
+        course = factories.Course(application_instance=application_instance)
```

Comment:
The test was creating a child grouping whose parent grouping has a different application instance: `factories.Course()` generates a new application instance for `course` (which is going to be used as the parent grouping) and then when the test calls `GroupingService.upsert_with_parent()`, `GroupingService` uses `ApplicationInstanceService.get_current()` as the application instance for the new grouping that it inserts.

The mock `application_instance_service`'s `get_current()` returns the value of the `application_instance` fixture. I might want to do something to remove this quite implicit dependency of this test on these connections between the three fixtures involved.